test/y2storage/boot_requirements_errors_test.rb

```diff
@@ -355,6 +355,18 @@
             include_examples "no errors"
           end
 
+          context "there is too big PReP partition" do
```

Comment:
Why is this tested only in the partitions-based case and not in LVM?

In fact, somebody could expect the check to behave differently in both scenarios. For partition-based installations in which several disks contain a PReP, we know which one is the relevant one (same disk than "/"), so we could fine-tune the check.

I'm not asking you to fine-tune the check, I'm just saying that it makes sense to include the test in both scenarios. In the first place to be better test-covered and in the second to specify that we don't do any extra check in the partition-based scenario.

Reply:
well, to be honest I do not get what is reason for different context as it is irrelevant to PREP proposal. in all case you need PReP that is directly on primary partition. So other tests are basically bogus that test identical stuff. There is also RAID and encrypted scenario and none of them make sense for non-power nv scenario. I am not sure if we really want complete matrix of various scenarios even that are not related to code. Because it start growing in too much directions.

Reply:
I can live without it, it's not a blocker for merging.

That being said "are not related to code" is exactly the mindset I usually try to fight when we talk about RSpec tests. I think about them from the perspective of the output they produce and the scenarios and behaviors that outputs describe. I don't think about the code when writing tests for a component. So this looked kind of confusing to me (it makes me wonder whether too big PReP is not checked with LVM):

```
with a partitions-based proposal
   there is a PReP partition
      does not contain an error
   there is too big PReP partition
      contains a warning for too big PReP partition
   PReP partition missing
      contains an error for missing PReP partition
with a LVM-based proposal
   there is a PReP partition
      does not contain an error
   PReP partition missing
      contains an error for missing PReP partition
```

In my opinion we think too much in terms of covering all the code and too little in terms of covering scenarios. But that's a philosophical topic for the workshop. As I said, I'm fine with the current tests, they are not a stopper.

Reply:
@ancorgs my problem with this approach is that for non-power PReP we should write something like "prep is always needed" and not writing all scenarios just to repeat that PReP is needed. It really does not help with reading output of rspec or documenting behavior. For me saying that prep is needed document also behavior not just code.